rosetta/transactor/transactor.go

```diff
@@ -0,0 +1,592 @@
+	if err != nil {
+		return nil, fmt.Errorf("invalid sender account: %w", err)
+	}
+	_, err = t.validate.Account(receive.AccountID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid receiver account: %w", err)
+	}
+
+	// Validate that the specified operations are transfers.
+	if send.Type != dps.OperationTransfer || receive.Type != dps.OperationTransfer {
+		return nil, failure.InvalidIntent{
+			Description: failure.NewDescription("only transfer operations are supported",
+				failure.WithString("withdrawal_type", send.Type),
+				failure.WithString("deposit_type", receive.Type),
+			),
+		}
+	}
+
+	amount := amounts[0]
+	if amount < 0 {
+		amount = -amount
+	}
+
+	intent := Intent{
+		From:     flow.HexToAddress(send.AccountID.Address),
+		To:       flow.HexToAddress(receive.AccountID.Address),
+		Amount:   cadence.UFix64(amount),
+		Payer:    flow.HexToAddress(send.AccountID.Address),
+		Proposer: flow.HexToAddress(send.AccountID.Address),
+		GasLimit: flow.DefaultMaxTransactionGasLimit,
+	}
+
+	return &intent, nil
+}
+
+// CompileTransaction creates a complete Flow transaction from the given intent and metadata.
+func (t *Transactor) CompileTransaction(rosBlockID identifier.Block, intent *Intent, sequence uint64) (string, error) {
+
+	// Generate script for the token transfer.
+	script, err := t.generate.TransferTokens(dps.FlowSymbol)
+	if err != nil {
+		return "", fmt.Errorf("could not generate transfer script: %w", err)
+	}
+
+	// TODO: Allow arbitrary proposal key index
+	// => https://github.com/optakt/flow-dps/issues/369
+
+	// Create the transaction.
+	unsignedTx := sdk.NewTransaction().
+		SetScript(script).
+		SetReferenceBlockID(sdk.HexToID(rosBlockID.Hash)).
+		SetPayer(sdk.Address(intent.Payer)).
+		SetProposalKey(sdk.Address(intent.Proposer), 0, sequence).
+		AddAuthorizer(sdk.Address(intent.From)).
+		SetGasLimit(intent.GasLimit)
+
+	receiver := cadence.NewAddress(flow.BytesToAddress(intent.To.Bytes()))
+
+	// Add the script arguments - the amount and the receiver.
+	_ = unsignedTx.AddArgument(intent.Amount)
+	_ = unsignedTx.AddArgument(receiver)
+
+	payload, err := t.encodeTransaction(unsignedTx)
+	if err != nil {
+		return "", fmt.Errorf("could not encode transaction: %w", err)
+	}
+
+	return payload, nil
+}
+
+func (t *Transactor) HashPayload(rosBlockID identifier.Block, unsigned string, signer identifier.Account) (string, error) {
+
+	unsignedTx, err := t.decodeTransaction(unsigned)
+	if err != nil {
+		return "", fmt.Errorf("could not decode transaction: %w", err)
+	}
+
+	// Validate block.
+	height, _, err := t.validate.Block(rosBlockID)
+	if err != nil {
+		return "", fmt.Errorf("could not validate block: %w", err)
+	}
+
+	// Validate address.
+	address, err := t.validate.Account(signer)
+	if err != nil {
+		return "", fmt.Errorf("could not validate account: %w", err)
+	}
+
+	key, err := t.invoke.Key(height, address, 0)
+	if err != nil {
+		return "", fmt.Errorf("could not retrieve key: %w", err)
```

Comment:
```suggestion
		return "", fmt.Errorf("could not retrieve key for address %s at height %d: %w", address.Hex(), height, err)
```

Reply:
The details are already contained in the wrapped error.